muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/guider/GuiderRESTService.java

```diff
@@ -688,6 +697,216 @@ private GuiderStudentWorkspaceActivityRestModel toRestModel(GuiderStudentWorkspa
         assessmentState);
     return model;
   }
+  /**
+   * POST mApi().guider.student.contactLog.create(userEntityId, payload)
+   * 
+   * Creates a new contact log entry from the given payload and studentEntityId.
+   * 
+   * Payload: 
+   * 
+   * {text: "something something",
+   *  entryDate: 2021-02-15
+   *  creatorName: "Etunimi Sukunomi",
+   *  type: "OTHER"/"LETTER"/"EMAIL"/"PHONE"/"CHATLOG"/"SKYPE"/"FACE2FACE"/"ABSENCE"/"MUIKKU"
+   * }
+   *  
+   * Output: Created student contact log entry
+   * 
+   * {
+   * id: 123;
+   * text: "something something",
+   * creatorId: 23,
+   * creatorName: "Etunimi Sukunimi";
+   * entryDate: 2021-02-15;
+   * type: "PHONE";
+   * }
+   */
+  @POST
+  @Path("/student/{ID}/contactLog/")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
+  public Response createStudentContactLog(@PathParam("ID") Long userEntityId, StudentContactLogEntryRestModel payload) {
+    String dataSource = sessionController.getLoggedUserSchoolDataSource();
+    UserEntity userEntity = userEntityController.findUserEntityById(userEntityId);
+    
+    BridgeResponse<StudentContactLogEntryRestModel> response = userSchoolDataController.createStudentContactLogEntry(dataSource, userEntity.defaultSchoolDataIdentifier(), payload);
+    if (response.ok()) {
+      return Response.status(response.getStatusCode()).entity(response.getEntity()).build();
+    }
+    else {
+      return Response.status(response.getStatusCode()).entity(response.getMessage()).build();
+    }
+  }
+  
+  /**
+   * PUT mApi().guider.student.contactLog.update(userEntityId, contactLogEntryId, payload)
+   * 
+   * payload: {
+   * text: "something something",
+   * entryDate: 2021-02-15;
+   * type: "PHONE";
+   * }
+   * @param userEntityId
+   * @param contactLogEntryId
+   * @param payload
+   * @return
+   * 
+   * Returns updated contact log entry
+   */
+  
+  @PUT
+  @Path("/student/{ID}/contactLog/{CONTACTLOGENTRYID}")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
+  public Response updateStudentContactLog(@PathParam("ID") Long userEntityId, @PathParam("CONTACTLOGENTRYID") Long contactLogEntryId, StudentContactLogEntryRestModel payload) {
+    String dataSource = sessionController.getLoggedUserSchoolDataSource();
+    UserEntity userEntity = userEntityController.findUserEntityById(userEntityId);
+    
+    BridgeResponse<StudentContactLogEntryRestModel> response = userSchoolDataController.updateStudentContactLogEntry(dataSource, userEntity.defaultSchoolDataIdentifier(), contactLogEntryId, payload);
```

Comment:
Extra safety by checking that the creatorId in the payload object is that of the logged in user?